### PR TITLE
Refactoring config parsing for common api

### PR DESCRIPTION
**Pull request by jasonsliu**

### CODE REVIEW

File: include/server.h [hunk 1]
```diff
@@ -5,11 +5,19 @@
 
 #include "connection.h"
 
+struct ServerInfo;
```
Comment (crystalhsieh) [inline include/server.h:8]: please make a comment of what will be inside ServerInfo

File: include/server.h [hunk 2]
```diff
@@ -5,11 +5,19 @@
...
+
 class Server {
  public:
-  Server(boost::asio::io_service& io_service, const char* filename);
+  // Factory method
+  static Server *makeServer(boost::asio::io_service& io_service, const char* config_file);
```
Comment (crystalhsieh) [inline include/server.h:13]: edit spacing so it's easier to identify "const char* config_file);" should be on the next line

File: src/server.cc [hunk 3]
```diff
@@ -54,3 +60,48 @@ bool Server::handle_accept(Connection::pointer new_connection,
...
+    std::string key = config.statements_[i]->tokens_[0];
+    if(key == "port") {
+      info->port = std::stoi(config.statements_[i]->tokens_[1]);
+    } else if (key == "path") {
+      if(config.statements_[i]->tokens_.size() != 3
+          || !config.statements_[i]->child_block_) {
+        return false;
+      }
+      // determine handler type
+      if(config.statements_[i]->tokens_[2] == "StaticFileHandler") {
```
Comment (crystalhsieh) [inline src/server.cc:81]: Since we haven't finished API implementation, these will change once we've finished, right?

**Review by crystalhsieh — COMMENTED** [review]
mostly nits, so looks good!

File: include/server.h (hunk 1 above)
Comment (jasonsliu) [reply include/server.h:8]: Wait it's defined below, we shouldn't need to comment here.
It's just to make Server happy.

File: include/server.h (hunk 2 above)
Comment (jasonsliu) [reply include/server.h:13]: Will do so in next pull request.

File: src/server.cc (hunk 3 above)
Comment (jasonsliu) [reply src/server.cc:81]: Yes, quite a bit!

**Pull request merged**